staging/src/k8s.io/legacy-cloud-providers/aws/aws_routes.go

```diff
@@ -127,13 +127,27 @@ func (c *Cloud) ListRoutes(ctx context.Context, clusterName string) ([]*cloudpro
 	return routes, nil
 }
 
-// Sets the instance attribute "source-dest-check" to the specified value
-func (c *Cloud) configureInstanceSourceDestCheck(instanceID string, sourceDestCheck bool) error {
-	request := &ec2.ModifyInstanceAttributeInput{}
-	request.InstanceId = aws.String(instanceID)
-	request.SourceDestCheck = &ec2.AttributeBooleanValue{Value: aws.Bool(sourceDestCheck)}
-
-	_, err := c.ec2.ModifyInstanceAttribute(request)
+// Sets the instance's primary network interface attribute "source-dest-check" to the specified value
+func (c *Cloud) configureInstanceSourceDestCheck(instance *ec2.Instance, sourceDestCheck bool) error {
+	instanceID := aws.StringValue(instance.InstanceId)
+	// If there are multiple network interfaces attached to the instance,
+	// the invocation of `ModifyInstanceAttributes()` would fail with the following error:
+	// "InvalidInstanceID: There are multiple interfaces attached to instance 'i-xx'. Please specify an interface ID for the operation instead."
+	// So we need to modify the attribute of primary network interface here.
+	var primaryInterfaceID *string
+	for _, device := range instance.NetworkInterfaces {
+		if aws.Int64Value(device.Attachment.DeviceIndex) == 0 {
+			primaryInterfaceID = device.NetworkInterfaceId
+			break
+		}
+	}
+	if primaryInterfaceID == nil {
+		return fmt.Errorf("no primary interface found on instance %s", instanceID)
+	}
+	_, err := c.ec2.ModifyNetworkInterfaceAttribute(&ec2.ModifyNetworkInterfaceAttributeInput{
```

Comment:
As per https://github.com/kubernetes/cloud-provider-aws/blob/master/README.md#iam-policy, I would expect this change might be breaking for some people because of IAM permissions. Might be a good idea to have the workflow same for single interface?

Reply:
This is a good point. updated as you said.